src/WebJobs.Script.WebHost/WebHostResolver.cs

```diff
@@ -1,329 +0,0 @@
-
-        public ISecretManager GetSecretManager() =>
-            GetSecretManager(_settings);
-
-        public ISecretManager GetSecretManager(WebHostSettings settings)
-        {
-            return GetWebScriptHostManager(settings).SecretManager;
-        }
-
-        public WebScriptHostManager GetWebScriptHostManager() =>
-            GetWebScriptHostManager(_settings);
-
-        public WebScriptHostManager GetWebScriptHostManager(WebHostSettings settings)
-        {
-            return GetActiveInstance(settings, ref _activeHostManager, ref _standbyHostManager);
-        }
-
-        /// <summary>
-        /// This method ensures that all services managed by this class are initialized
-        /// correctly taking into account specialization state transitions.
-        /// </summary>
-        internal void EnsureInitialized(WebHostSettings settings)
-        {
-            // Create a logger that we can use when the host isn't yet initialized.
-            ILogger logger = _loggerFactory.CreateLogger(LogCategories.Startup);
-
-            lock (_syncLock)
-            {
-                // Determine whether we should do normal or standby initialization
-                if (!WebScriptHostManager.InStandbyMode)
-                {
-                    // We're not in standby mode. There are two cases to consider:
-                    // 1) We _were_ in standby mode and now we're ready to specialize
-                    // 2) We're doing non-specialization normal initialization
-                    if (_activeHostManager == null &&
-                        (_standbyHostManager == null || _settingsManager.ContainerReady))
-                    {
-                        _specializationTimer?.Dispose();
-                        _specializationTimer = null;
-
-                        _activeScriptHostConfig = CreateScriptHostConfiguration(settings);
-                        _activeHostManager = new WebScriptHostManager(_activeScriptHostConfig, _secretManagerFactory, _eventManager, _settingsManager, settings,
-                            _router, loggerProviderFactory: _loggerProviderFactory, loggerFactory: _loggerFactory, eventGenerator: _eventGenerator);
-                        //_activeReceiverManager = new WebHookReceiverManager(_activeHostManager.SecretManager);
-                        InitializeFileSystem(settings, _settingsManager.FileSystemIsReadOnly);
-
-                        if (_standbyHostManager != null)
-                        {
-                            // we're starting the one and only one
-                            // standby mode specialization
-                            logger.LogInformation(Resources.HostSpecializationTrace);
-
-                            // After specialization, we need to ensure that custom timezone
-                            // settings configured by the user (WEBSITE_TIME_ZONE) are honored.
-                            // DateTime caches timezone information, so we need to clear the cache.
```

Comment:
ALL the logic in this method has found a new home? A bunch of tricky stuff in here and lots of opportunity for regressions

Reply:
E.g. where did the TimeZoneInfo.ClearCachedData end up?